src/tribler/core/utilities/tests/test_tracker_utils.py

```diff
@@ -6,137 +6,74 @@
-    assert not result
-
-
-def test_uniform_bad_urlenc():
-    result = get_uniformed_tracker_url("http://btjunkie.org/?do=upload")
-    assert not result
-
-
-def test_uniform_empty():
-    result = get_uniformed_tracker_url('')
-    assert not result
-
-
-def test_skip_truncated_url():
-    result = get_uniformed_tracker_url("http://tracker.1337x.org:80/anno...")
-    assert not result
-
-
-def test_skip_wrong_url_scheme():
-    result = get_uniformed_tracker_url("wss://tracker.1337x.org:80/announce")
-    assert not result
-
-
-def test_skip_value_error():
-    result = get_uniformed_tracker_url("ftp://tracker.1337\xffx.org:80/announce")
-    assert not result
-
-
-def test_skip_split_error():
-    result = get_uniformed_tracker_url(";")
-    assert not result
-
-
-def test_parse_scheme_correct_udp():
-    result = parse_tracker_url("udp://tracker.openbittorrent.com:80")
-    assert result == ("udp", ("tracker.openbittorrent.com", 80), '')
-
-
-def test_parse_scheme_correct_http():
-    result = parse_tracker_url("http://torrent.ubuntu.com:6969/announce")
-    assert result == ("http", ("torrent.ubuntu.com", 6969), "/announce")
-
-
-def test_parse_scheme_unknown():
-    with pytest.raises(MalformedTrackerURLException):
-        parse_tracker_url("unknown://ipv6.torrent.ubuntu.com:6969/announce")
-
-
-def test_parse_bad_url():
-    with pytest.raises(MalformedTrackerURLException):
-        parse_tracker_url("http://foo.com:6969/announce( %(")
-
-
-def test_parse_scheme():
-    with pytest.raises(MalformedTrackerURLException):
-        parse_tracker_url("http://torrent.ubuntu.com:80")
-
-
-def test_parse_http_no_announce_path():
+TRACKER_HOST = 'tracker.example.com'
+
+EXPECTED_UNIFORM_URLS = [
+    (f'udp://{TRACKER_HOST}:80', f'udp://{TRACKER_HOST}:80'),  # Proper UDP URL
+    (f'udp://{TRACKER_HOST}', None),  # UDP no port
+    (f'udp://{TRACKER_HOST}:6969/announce', f'udp://{TRACKER_HOST}:6969'),  # UDP remove path
+    (f'udp://{TRACKER_HOST}:80\x00', f'udp://{TRACKER_HOST}:80'),  # Trailing zero hex
+    (f'udp://{TRACKER_HOST}:80\xff', None),  # Trailing non-zero hex
+
+    (f'http://{TRACKER_HOST}:6969/announce', f'http://{TRACKER_HOST}:6969/announce'),  # Correct HTTP URL
+    (f'http://{TRACKER_HOST}:6969/announce/', f'http://{TRACKER_HOST}:6969/announce'),  # Trailing /
+    (f'http://{TRACKER_HOST}', None),  # HTTP no path
+    (f'http://{TRACKER_HOST}\xa0', None),  # HTTP no path nbsp
+    (f'http://{TRACKER_HOST}/announce', f'http://{TRACKER_HOST}/announce'),  # HTTP default port
+    (f'http://{TRACKER_HOST}:80/announce', f'http://{TRACKER_HOST}/announce'),  # HTTP default port given
+    (f'http://{TRACKER_HOST}/?do=upload', None),  # Bad URL encoding
+    (f'http://{TRACKER_HOST}:80/anno...', None),  # Truncated URL
+
+    (f'https://{TRACKER_HOST}/announce', f'https://{TRACKER_HOST}/announce'),  # HTTPS default port
+    (f'https://{TRACKER_HOST}:443/announce', f'https://{TRACKER_HOST}/announce'),  # HTTPS default port given
```

Comment:
> Why is this code necessary? Can you provide an example of the incorrect URL that this code fixes? (I suspect the fixed URL is not entirely correct and can include an excess /, but may be wrong on that)

@kozlovsky As in this test, if the default port is given, then it is removed from the expected uniform URL.

Reply:
Ah, ok, I misread it